### ISSUE TITLE
Execute batch requests internally

### ISSUE THREAD

**Issue by john-doherty** [john-doherty/express-batch-requests#1]
At present, this module uses the npm `request` module to request each endpoint. This causes an unnecessary external HTTP request - it should really resolve internally.

**Comment by iNDicat0r** [comment]
I agree with the above, just use the `router.handle` to dispatch, otherwise its a poor design!

**Comment by iNDicat0r** [comment]
@john-doherty take a look at: https://github.com/Aminadav/node-run-middleware

**Comment by john-doherty** [comment]
@iNDicat0r this wont work as it requires the consuming app to use `runMiddleware(app)` as their first middleware. It uses this method to capture all the routes.

I created [express-request-transfer](https://github.com/john-doherty/express-request-transfer) that should help solve this problem, just need to find the time to update
